### ISSUE TITLE
Lint/AssignmentInCondition

### ISSUE THREAD

**Issue by Linuus** [rubocop-hq/rubocop#7348]
## Expected behavior

There should be a warning about the code below when `AllowSafeAssignment` is set to false.

```
times = 10
count = 0
if (count += 1) <= times
  retry
end
```

## Actual behavior

It does not output a warning.

## Steps to reproduce the problem
Add:
```
Lint/AssignmentInCondition:
  AllowSafeAssignment: false
```
In your `.rubocop.yml` and run rubocop on a file containing:

```ruby
times = 10
count = 0
if (count += 1) <= times
  retry
end
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ [bundle exec] rubocop -V
0.71.0 (using Parser 2.6.3.0, running on ruby 2.5.5 x86_64-darwin17)
```

The same thing happens with `0.74.0`.

**Comment by koic** [comment]
I think this is the expected behavior, it is not a false negative.

I think the purpose of this cop is to check whether `=` is used by mistake where `==` should be used. `==` and `=` are visually similar and feel suspicion whether it is a typo. OTOH, `+=` is unique as a self-assignment expression.
